rust/bridge/node/futures/src/executor.rs

```diff
@@ -0,0 +1,110 @@
+//
+// Copyright 2021 Signal Messenger, LLC.
+// SPDX-License-Identifier: AGPL-3.0-only
+//
+
+use neon::prelude::*;
+use std::future::Future;
+use std::pin::Pin;
+use std::sync::{Arc, Mutex};
+use std::task::{Poll, Wake};
+
+/// Adds support for executing futures on a Neon [EventQueue][].
+///
+/// [EventQueue]: https://docs.rs/neon/0.7.0-napi.3/neon/event/struct.EventQueue.html
+pub trait EventQueueEx {
+    /// Schedules the future to run on the JavaScript main thread until complete.
+    fn send_future(&self, future: impl Future<Output = ()> + 'static + Send);
+}
+
+impl EventQueueEx for EventQueue {
+    fn send_future(&self, future: impl Future<Output = ()> + 'static + Send) {
+        self.send(move |mut cx| {
+            let task = Arc::new(FutureTask {
+                queue: cx.queue(),
+                future: Mutex::new(Some(Box::pin(future))),
+            });
+            task.poll();
+            Ok(())
+        })
+    }
+}
+
+/// Used to "send" a task from a thread to itself through a multi-threaded interface.
+struct AssertSendSafe<T>(T);
```

Comment:
The purpose here is to support Futures that *aren't* Send. We know it's safe in this case because we're on the queue when we start and we're sending the future to the same queue.

It's possible we'll never end up using this with anything that's not Send, though, in which case I'm sympathetic to the idea of "it's unnecessary `unsafe`ty". But if we do need it, I wouldn't want to go back and work it out all over again.